homeassistant/components/climate/__init__.py

```diff
@@ -219,6 +237,32 @@ def away_mode_set_service(service):
         descriptions.get(SERVICE_SET_AWAY_MODE),
         schema=SET_AWAY_MODE_SCHEMA)
 
+    def home_mode_set_service(service):
+        """Set home mode on target climate devices."""
+        target_climate = component.extract_from_service(service)
+
+        home_mode = service.data.get(ATTR_HOME_MODE)
+
+        if home_mode is None:
```

Comment:
Regarding the "this will never happen". The same check are already in the code for "away_mode_set_service", so for consistency I would like retain it. Also, I found that services crash when one is missing parameters so I am not sure about your comment regarding "this will never happen". See issue #4871.